frontend/src/ui/Select/Select.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from 'react'
+import Autocomplete from '@mui/joy/Autocomplete'
+import Box from '@mui/joy/Box'
+import { styled } from '@mui/joy/styles'
+import { AutocompleteProps } from '@mui/joy/Autocomplete'
+
+type Option = {
+  value: string
+  label: string
+}
+
+type SelectProps = {
+  label: string
+  value?: string
+} & Omit<AutocompleteProps<Option, undefined, undefined, undefined>, 'value'>
```

Comment:
What do these undefined mean here?

Reply:
They come from the `AutocompleteProps`:

```ts
interface AutocompleteComponent {
    <T, Multiple extends boolean | undefined = undefined, DisableClearable extends boolean | undefined = undefined, FreeSolo extends boolean | undefined = undefined>(props: AutocompleteProps<T, Multiple, DisableClearable, FreeSolo>): JSX.Element;
    propTypes?: any;
}
```

We have the generic `T` (which is the `Option` type in our case), and then we have: Multiple, DisableClearable and FreeSolo, but we aren't using any of those props in the component, so they are all undefined.

Reply:
here is the signature in the .d.ts file:
```ts
/**
 * AutocompleteProps signature:
 * @template T
 * @param {string | object} T The option structure, must be a string or an object (by default, only accepts objects with { label: string } )
 * @param {boolean | undefined} Multiple If your component is set with property multiple as true
 * @param {boolean | undefined} DisableClearable If your component is set with property disableClearable as true
 * @param {boolean | undefined} FreeSolo If your component is set with property freeSolo as true
 */
 ```

Reply:
I think it could also be: `AutocompleteProps<Option, boolean, boolean, boolean>` but I don't think it would change anything, I'm just taking those props to extend it to the `Select` component so we can use the same API

Reply:
Got it, thank you!